Add a Tutorial for Integration Testing with khttp

kevincianfarini opened the pull request:
In this tutorial I go through setting up a very limited CRUD API, how to run it from a Test Runner, and making requests with `khttp`. Let me know what you think!

tipsy commented on _posts/tutorials/2018-03-01-javalin-kotlin-integration-testing-example.md:
```diff
@@ -0,0 +1,254 @@
+
+class JavalinApp(private val port: Int) {
+
+    fun create(): Javalin {
```
Perhaps call this `init()` or `start()` to avoid confusion? `Javalin.create()` just creates a server.

kevincianfarini commented:
I can check again in a bit, but when I was passing a `FoodItem` instance to ctx.json it was just calling it's `toString`. Maybe this has something to do with kotlin data classes?

tipsy commented:
Did you use the Kotlin module?
```
// https://mvnrepository.com/artifact/com.fasterxml.jackson.module/jackson-module-kotlin
compile group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version: '2.9.4.1'
```

kevincianfarini commented:
The wrong gradle dependency was the culprit. I'll update the tutorial and PR accordingly. 

Also I changed `.create()` to `.init()`

kevincianfarini commented:
Commit has been amended. Removed dependency to GSON. Either use `ctx.json` or an extension to deserialize that depends on `Jackson`.

Removed unnecessary `toString` serialization.

tipsy commented:
Thanks @kevincianfarini, looks great!

kevincianfarini commented:
Seems as though the site isn't picking up code colouring. Is that intended?

tipsy commented:
You just have to specify the language. Pretend that the single-quotes are backticks:

```
'''kotlin
    code
'''
```

tipsy commented:
I can fix it when I come back from the store.

tipsy commented:
Fixed.
